core-plugins/src/test/java/io/cdap/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -982,6 +1001,41 @@ public void testFileBatchInputFormatParquetMissingField() throws Exception {
     Assert.assertEquals(expected, output);
   }
 
+  @Test
+  public void testDelimitedFormatsValidSchema() throws Exception {
+    FileSourceConfig config = new FileSourceConfig();
+    FailureCollector collector = new MockFailureCollector();
+
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("referenceName"), "ref");
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("format"), "csv");
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("schema"),
+                         RECORD_SCHEMA_VALID_DATE.toString());
+    FieldSetter.setField(config, FileSourceConfig.class.getDeclaredField("path"), "path");
+
+    config.validate(collector);
+    Assert.assertEquals(0, collector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testDelimitedFormatsInValidSchema() throws Exception {
+    FileSourceConfig config = new FileSourceConfig();
+    FailureCollector collector = new MockFailureCollector();
+
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("referenceName"), "ref");
+    FieldSetter.setField(config, AbstractFileSourceConfig.class.getDeclaredField("format"), "csv");
```

Comment:
The format we are trying to test is `delimited`, not `csv`, right?

Reply:
PR updated.